code/game/objects/items/devices/instruments.dm

```diff
@@ -268,3 +268,39 @@
 	item_state = "electronic"
 	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+
+/obj/item/musicbeacon
+	name = "express delivery beacon"
+	desc = "Summon your tool of art."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "gangtool-red"
+	item_state = "radio"
+
+/obj/item/musicbeacon/attack_self(mob/user)
+	beacon_music(user)
+
+/obj/item/musicbeacon/proc/beacon_music(mob/M)
+	var/list/instruments = list(
+								/obj/item/instrument/violin,
+								/obj/item/instrument/piano_synth,
+								/obj/item/instrument/guitar,
+								/obj/item/instrument/eguitar,
+								/obj/item/instrument/glockenspiel,
+								/obj/item/instrument/accordion,
+								/obj/item/instrument/trumpet,
+								/obj/item/instrument/saxophone,
+								/obj/item/instrument/trombone,
+								/obj/item/instrument/recorder,
+								/obj/item/instrument/harmonica
+								)
+	var/list/display_names = list()
+	for(var/V in instruments)
+		var/atom/A = V
+		display_names += list(initial(A.name) = A)
+	var/choice = input(M,"What instrument would you like to order?","Jazz Express") as null|anything in display_names
+	if(!M.canUseTopic(src, BE_CLOSE) || !choice)
+		return
+	var/instrument = new display_names[choice]
+	if(instrument)
+		qdel(src)
+	M.put_in_active_hand(instrument)

```

Comment:
You're operating under the assumption here that the instrument may not exist, yet if it doesn't new properly you're still trying to add it (null) to the active hand. Just remove the `if(instrument)` check and let it runtime if it fails for some reason.